crates/bevy_reflect/src/lib.rs

```diff
@@ -64,6 +64,176 @@ pub use type_uuid::*;
 pub use bevy_reflect_derive::*;
 pub use erased_serde;
 
+use thiserror::Error;
+
+/// An Error for failed conversion of reflected type to original type in [`FromReflect::from_reflect`].
+///
+/// Within variants `FieldError`, `IndexError`, `VariantError`, `KeyError` and `ValueError`;
+/// [`Error::source`](std::error::Error::source) must be used to trace the underlying error.
+#[derive(Error, Debug)]
+pub enum FromReflectError {
```

Comment:
If we put this in `from_reflect` module, then all the instances outside of `from_reflect` module won't be able create the Error in the first place since all its fields are private.

Or else we would have to make them public. Is this what was intended?

Reply:
I don't think enum variant fields can be made private? We should be able to define it in the module and allow it to be exported by the `pub use from_reflect::*` in `lib.rs`.